src/instana/agent/host.py

```diff
@@ -297,73 +324,147 @@ def report_data_payload(self, payload):
         except urllib3.exceptions.MaxRetryError:
             pass
         except Exception as exc:
-            logger.debug("report_data_payload: Instana host agent connection error (%s)", type(exc), exc_info=True)
+            logger.debug(
+                f"report_data_payload: Instana host agent connection error ({type(exc)})",
+                exc_info=True,
+            )
         return response
 
-    def handle_agent_tasks(self, task):
+    def report_metrics(self, payload: Dict[str, Any]) -> Union[Response, None]:
```

Comment:
Use `Optional` instead of `Union[None, something]`.
https://docs.python.org/3.13/library/typing.html#typing.Optional 

```suggestion
    def report_metrics(self, payload: Dict[str, Any]) -> Optional[Response]:
```